transport/src/main/java/io/netty/channel/VoidChannelPromise.java

```diff
@@ -21,7 +21,7 @@
 
 import java.util.concurrent.TimeUnit;
 
-final class VoidChannelPromise extends AbstractFuture<Void> implements ChannelPromise {
+public final class VoidChannelPromise extends AbstractFuture<Void> implements ChannelPromise {
```

Comment:
made this one public to be able to use it in the custom `Channel` implementation